From eee2ecf859027bb680e96f6bb43dc57fa36e695b Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Mon, 27 Jan 2025 15:32:57 +0100
Subject: [PATCH 1574/1695] drivers: stm32_exti: use standard DT interrupt type

Drop using EXTI specific interrupt types and reuse the types
already defined in dt-bindings/interrupt-controller/irq.h

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 9cf1697bb287277ad47bffd9b4be207538433081 drivers: stm32_exti: use standard DT interrupt type]
Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: I12b00a14aa5581f9ed8905eb2ebfaf284a95f91f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/434921
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/drivers/stm32_exti.c         | 8 +++++---
 core/include/drivers/stm32_exti.h | 4 ----
 2 files changed, 5 insertions(+), 7 deletions(-)

diff --git a/core/drivers/stm32_exti.c b/core/drivers/stm32_exti.c
index e4926e690..ef73817ba 100644
--- a/core/drivers/stm32_exti.c
+++ b/core/drivers/stm32_exti.c
@@ -5,6 +5,7 @@
 
 #include <drivers/stm32_exti.h>
 #include <drivers/stm32_rif.h>
+#include <dt-bindings/interrupt-controller/irq.h>
 #include <io.h>
 #include <kernel/boot.h>
 #include <kernel/dt.h>
@@ -126,19 +127,20 @@ void stm32_exti_set_type(struct stm32_exti_pdata *exti, uint32_t exti_line,
 	unsigned int i = 0;
 
 	switch (type) {
-	case EXTI_TYPE_RISING:
+	case IRQ_TYPE_EDGE_RISING:
 		r_trig |= mask;
 		f_trig &= ~mask;
 		break;
-	case EXTI_TYPE_FALLING:
+	case IRQ_TYPE_EDGE_FALLING:
 		r_trig &= ~mask;
 		f_trig |= mask;
 		break;
-	case EXTI_TYPE_BOTH:
+	case IRQ_TYPE_EDGE_BOTH:
 		r_trig |= mask;
 		f_trig |= mask;
 		break;
 	default:
+		EMSG("Unsupported interrupt type 0x%"PRIx32, type);
 		panic();
 	}
 
diff --git a/core/include/drivers/stm32_exti.h b/core/include/drivers/stm32_exti.h
index 07a246c45..3144ccab7 100644
--- a/core/include/drivers/stm32_exti.h
+++ b/core/include/drivers/stm32_exti.h
@@ -8,10 +8,6 @@
 
 #include <tee_api_types.h>
 
-#define EXTI_TYPE_RISING	1
-#define EXTI_TYPE_FALLING	2
-#define EXTI_TYPE_BOTH	(EXTI_TYPE_RISING | EXTI_TYPE_FALLING)
-
 struct stm32_exti_pdata;
 
 /*
-- 
2.47.2

